### DISCUSSION TITLE
Conditional exception-free getters

### DISCUSSION THREAD

**Discussion by sergeniously** [nlohmann/json#4090] [Ideas]
Does the library have standard way (API?) to extract value only when a key exists (without throwing an exception)?
Something like this:
```c++
template<class Value>
bool unjson(const nlohmann::json& json, const char* key, Value& value) {
    if (auto it = json.find(key); it != json.end()) {
        it->get_to(value);
        return true;
    }
    return false;
}
```

**Comment by nlohmann** [comment]
There is the value() function, and contains(). See https://json.nlohmann.me/api/basic_json/

**Reply by sergeniously** [reply]
Those are pretty inconvenient and lead to a redundancy when we say about extracting a structure's members.
Take a look, for example:
```c++
struct User {
  uint64_t id = 0;
  string name = "anonym";

  void fromJson(const nlohmann::json& json) {
    if (false) {
      // redundant code
      id = json.value("id", id);
      name = json.value("name", name);
    } else {
      // good looking code (members already have default values)
      unjson(json, "id", id);
      unjson(json, "name", name);
  }
};
```

It would be nice to have an API e.g. **value_to** with that logic...

**Reply by nlohmann** [reply]
Can't you build this with contains()?

**Reply by sergeniously** [reply]
Oh... well, it seems I'll have to use my own wrappers.
